pkg/asset-manager-utils/contracts/aave/SinglePoolAaveATokenAssetManager.sol

```diff
@@ -0,0 +1,119 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import "./ILendingPoolAddressesProvider.sol";
+import "./ILendingPool.sol";
+import "./IAaveIncentivesController.sol";
+
+import "../SinglePoolAssetManager.sol";
+import "@balancer-labs/v2-distributors/contracts/interfaces/IMultiRewards.sol";
+
+pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
+
+// solhint-disable no-empty-blocks
+// solhint-disable var-name-mixedcase
+// solhint-disable private-vars-leading-underscore
+contract SinglePoolAaveATokenAssetManager is SinglePoolAssetManager {
+    uint16 private constant REFERRAL_CODE = 0;
+
+    IAaveIncentivesController public immutable aaveIncentives;
+    ILendingPool public immutable lendingPool;
+    IERC20 public immutable aToken;
+    IERC20 public immutable stkAave;
+
+    /// @notice staking contract for pool which owns this asset manager
+    IMultiRewards public distributor;
+
+    constructor(
+        IVault _vault,
+        address _token,
+        address _lendingPool,
+        address _aToken,
+        address _aaveIncentives,
+        address _stkAave
+    ) SinglePoolAssetManager(_vault, bytes32(0), _token) {
+        // TODO: pull these from Aave addresses provider
+        lendingPool = ILendingPool(_lendingPool);
+        aToken = IERC20(_aToken);
+
+        aaveIncentives = IAaveIncentivesController(_aaveIncentives);
+        stkAave = IERC20(_stkAave);
+
+        IERC20(_token).approve(_lendingPool, type(uint256).max);
+    }
+
+    /**
+     * @dev Should be called in same transaction as deployment through a factory contract
+     * @param pId - the id of the pool
+     * @param rewardsContract - the address of the rewards contract (to distribut stkAAVE)
+     */
+    function initialise(bytes32 pId, address rewardsContract) public {
+        require(poolId == bytes32(0), "Already initialised");
+        poolId = pId;
+        distributor = IMultiRewards(rewardsContract);
+        IERC20(stkAave).approve(rewardsContract, type(uint256).max);
+    }
+
+    /**
+     * @dev To be used in the case of a staking contract migration
+     */
+    function updateStakingContract(address rewardsContract) external {
+        require(msg.sender == address(distributor), "Must be called by current staking contract");
```

Comment:
It seems like we're using three different names here ('staking contract', 'rewards contract' and 'distributor') to refer to the same entity - we should pick a single one. 

Out of these, I'd go with either 'rewards contract' or 'distributor' (maybe 'rewards distributor' is the most clear one), as 'staking' refers to an implementation detail and not the goal of that component.